### PR TITLE
docs: change log for v4.2.4bp2

### PR DESCRIPTION
**Pull request by yhilmare**
#### What type of PR is this?

type-docs

#### What this PR does / why we need it:
 add changelog
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: CHANGELOG-zh-CN.md [hunk 1]
```diff
@@ -1,6 +1,89 @@
...
+
+系统集成
+
+- 第三方用户集成中，用户如果修改了 extra_properties 将无法再登录 ODC [#2336](https://github.com/oceanbase/odc/pull/2336)
+- 同时使用 OAuth 和堡垒机集成时无法打开 SQL 控制台 [#2253](https://github.com/oceanbase/odc/pull/2253)
+
+其他
+
+- 生命周期横跨发布过程的工单无法正常推进 [#2065](https://github.com/oceanbase/odc/pull/2065)
+- 无法手动执行活终止任务 [#2272](https://github.com/oceanbase/odc/pull/2272)
```
Comment (yizhouxw) [inline CHANGELOG-zh-CN.md:54]: 活 --> 或

File: CHANGELOG-zh-CN.md [hunk 2]
```diff
@@ -1,6 +1,89 @@
...
+- 输入较长 SQL 时，无法完全回显语句 [#2353](https://github.com/oceanbase/odc/pull/2353)
+
+数据库归档/清理
+
+- 任务被终止时状态被错误地设置为成功 [#2340](https://github.com/oceanbase/odc/pull/2340)
+- OceanBase MySQL 归档自定义表名报表不存在问题 [#2313](https://github.com/oceanbase/odc/pull/2313)
+
+结果集导出
+
+- 导出结果集时发生超时 [#2315](https://github.com/oceanbase/odc/pull/2315)
```
Comment (yizhouxw) [inline CHANGELOG-zh-CN.md:34]: 数据库访问较慢时导出结果集会因为超时而失败

File: CHANGELOG-zh-CN.md [hunk 3]
```diff
@@ -1,6 +1,89 @@
...
+
+- 存在重名的无效数据库时，脱敏失效 [#2385](https://github.com/oceanbase/odc/pull/2385)
+
+命令行窗口
+
+- 输入较长 SQL 时，无法完全回显语句 [#2353](https://github.com/oceanbase/odc/pull/2353)
+
+数据库归档/清理
+
+- 任务被终止时状态被错误地设置为成功 [#2340](https://github.com/oceanbase/odc/pull/2340)
```
Comment (yizhouxw) [inline CHANGELOG-zh-CN.md:29]: 任务被终止时状态可能被错误地设置为成功

File: CHANGELOG-zh-CN.md [hunk 4]
```diff
@@ -1,6 +1,89 @@
...
+
+- 无法连接 ODP-Sharding 数据源 [#2339](https://github.com/oceanbase/odc/pull/2339)
+
+用户管理
+
+- 无法删除已归档项目的 OWNER 或 DBA 用户 [#2359](https://github.com/oceanbase/odc/pull/2359)
+
+影子表同步
+
+- 生成的语句语法错误 [#2388](https://github.com/oceanbase/odc/pull/2388)
```
Comment (yizhouxw) [inline CHANGELOG-zh-CN.md:17]: 列存在默认值时生成的语句可能出现语法错误

File: CHANGELOG-zh-CN.md [hunk 5]
```diff
@@ -1,6 +1,89 @@
...
+
+### 缺陷修复
+
+PL 调试
+
+- PL 调试超时参数无法通过连接初始化脚本设置 [#2179](https://github.com/oceanbase/odc/pull/2179)
+
+其他
+
+- 应用启动失败 [#2158](https://github.com/oceanbase/odc/pull/2158)
```
Comment (yizhouxw) [inline CHANGELOG-zh-CN.md:80]: 存在分区计划历史任务时 ODC Server 启动失败

File: CHANGELOG-zh-CN.md [hunk 6]
```diff
@@ -1,6 +1,89 @@
...
+- 在 OceanBase 4.x 版本下漏删分区以及少建分区 [#2327](https://github.com/oceanbase/odc/pull/2327)
+- 无法在 OceanBase 3.x 以下的版本发起分区计划任务 [#2323](https://github.com/oceanbase/odc/pull/2323)
+- 分区计划详情中不显示分区间隔
+
+系统集成
+
+- 第三方用户集成中，用户如果修改了 extra_properties 将无法再登录 ODC [#2336](https://github.com/oceanbase/odc/pull/2336)
+- 同时使用 OAuth 和堡垒机集成时无法打开 SQL 控制台 [#2253](https://github.com/oceanbase/odc/pull/2253)
+
+其他
```
Comment (yizhouxw) [inline CHANGELOG-zh-CN.md:51]: add 2
- 运行一段时间之后无法使用帐密登陆需要重启 ODCServer 才能恢复 [#2389](https://github.com/oceanbase/odc/pull/2389)
- Basic 认证方式调用 API 和 CSRF 防护存在冲突 [#2370](https://github.com/oceanbase/odc/pull/2370)

File: CHANGELOG-zh-CN.md (hunk 1 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:54]: 😅 done

File: CHANGELOG-zh-CN.md (hunk 2 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:34]: done

File: CHANGELOG-zh-CN.md (hunk 3 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:29]: done

File: CHANGELOG-zh-CN.md (hunk 4 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:17]: done

File: CHANGELOG-zh-CN.md (hunk 5 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:80]: done

File: CHANGELOG-zh-CN.md (hunk 6 above)
Comment (yhilmare) [reply CHANGELOG-zh-CN.md:51]: added

**Review by yizhouxw — DISMISSED** [review]
lgtm

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,4 +1,89 @@
...
+- When there is an invalid database with the same name, desensitization fails [#2385](https://github.com/oceanbase/odc/pull/2385)
+
+Command line window
+
+- When entering a long SQL, the statement cannot be fully echoed [#2353](https://github.com/oceanbase/odc/pull/2353)
+
+Database archiving/cleaning
+
+- The status may be incorrectly set to success when a task is terminated [#2340](https://github.com/oceanbase/odc/pull/2340)
+- There is no problem with OceanBase MySQL archiving custom table name reports [#2313](https://github.com/oceanbase/odc/pull/2313)
```
Comment (JessieWuJiexi) [inline CHANGELOG.md:28]: If the table names are different between the source and target when configuring data archiving tasks, an error occurs stating that the table does not exist.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,4 +1,89 @@
-## 4.2.4 (2024-04-02)
+## 4.2.4_bp2 (2024-05-14)
+
+### Bug fixes
+
+Data source
+
+- Unable to connect to ODP-Sharding data source [#2339](https://github.com/oceanbase/odc/pull/2339)
```
Comment (JessieWuJiexi) [inline CHANGELOG.md:7]: 建议所有的句子末尾都加上句号

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,4 +1,89 @@
...
+
+User Management
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:9]: done

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,4 +1,89 @@
...
+
+- Unable to delete OWNER or DBA user of archived project [#2359](https://github.com/oceanbase/odc/pull/2359)
+
+Shadow table synchronization
+
+- Syntax errors may occur in statements generated when a column has a default value [#2388](https://github.com/oceanbase/odc/pull/2388)
+
+Data Security
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:17]: done

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,4 +1,89 @@
...
+
+- When there is an invalid database with the same name, desensitization fails [#2385](https://github.com/oceanbase/odc/pull/2385)
+
+Command line window
+
+- When entering a long SQL, the statement cannot be fully echoed [#2353](https://github.com/oceanbase/odc/pull/2353)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:23]: done

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,4 +1,89 @@
...
+
+Database archiving/cleaning
+
+- The status may be incorrectly set to success when a task is terminated [#2340](https://github.com/oceanbase/odc/pull/2340)
+- There is no problem with OceanBase MySQL archiving custom table name reports [#2313](https://github.com/oceanbase/odc/pull/2313)
+
+ResultSet export
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:30]: done

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,4 +1,89 @@
...
+
+- When database access is slow, exporting the result set fails due to timeout [#2315](https://github.com/oceanbase/odc/pull/2315)
+
+Notification
+
+- Editing the message channel will cause the signing key to be lost [#2314](https://github.com/oceanbase/odc/pull/2314)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:36]: done

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,4 +1,89 @@
...
+
+Partition plan
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:38]: done

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,4 +1,89 @@
...
+
+- Partitions are missing and partitions are not created in OceanBase 4.x version [#2327](https://github.com/oceanbase/odc/pull/2327)
+- Unable to initiate partition scheduled tasks in versions below OceanBase 3.x [#2323](https://github.com/oceanbase/odc/pull/2323)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:41]: done

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,4 +1,89 @@
...
+- Partition intervals are not shown in partition plan details
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:42]: done

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,4 +1,89 @@
...
+
+System integration
+
+- In third-party user integration, users who modify extra_properties will no longer be able to log in to ODC [#2336](https://github.com/oceanbase/odc/pull/2336)
+- SQL console cannot be opened when using OAuth and bastion host integration at the same time [#2253](https://github.com/oceanbase/odc/pull/2253)
+
+Other
+
+- Work orders whose life cycle spans the release process cannot be promoted normally [#2065](https://github.com/oceanbase/odc/pull/2065)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:51]: done

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,4 +1,89 @@
...
+- Unable to execute or terminate tasks manually [#2272](https://github.com/oceanbase/odc/pull/2272)
+- After running for a period of time, you cannot use the account password to log in and you need to restart ODCServer to recover [#2389](https://github.com/oceanbase/odc/pull/2389)
+- There is a conflict between Basic authentication method calling API and CSRF protection [#2370](https://github.com/oceanbase/odc/pull/2370)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:54]: done

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,4 +1,89 @@
...
+
+### Usability improvements
+
+- The partition plan adds copywriting tips for the case where no partition pre-creation statement or partition deletion statement is generated [#2351](https://github.com/oceanbase/odc/pull/2351)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:58]: done

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,4 +1,89 @@
...
+- OceanBase versions before 4.2 disable full-link diagnostics [#2219](https://github.com/oceanbase/odc/pull/2219)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:59]: done

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,4 +1,89 @@
...
+
+### Dependency library upgrade
+
+- Upgrade data-lifecycle-manager version to 1.1.1 [#2281](https://github.com/oceanbase/odc/pull/2281)
+
+### Security
+
+- Remove snappy-java dependency [#2317](https://github.com/oceanbase/odc/pull/2317)
+- Data desensitization increases verification to avoid DDos risks caused by BigDecimal [#2271](https://github.com/oceanbase/odc/pull/2271)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:68]: done

File: CHANGELOG.md [hunk 22]
```diff
@@ -1,4 +1,89 @@
...
+
+## 4.2.4_bp1 (2024-04-12)
+
+### Bug fixes
+
+PL debugging
+
+- PL debug timeout parameter cannot be set via connection initialization script [#2179](https://github.com/oceanbase/odc/pull/2179)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:76]: done

File: CHANGELOG.md [hunk 23]
```diff
@@ -1,4 +1,89 @@
...
+
+Other
+
+- ODC Server fails to start when there is a partition plan history task [#2158](https://github.com/oceanbase/odc/pull/2158)
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply CHANGELOG.md:80]: done

File: CHANGELOG.md (hunk 7 above)
Comment (yhilmare) [reply CHANGELOG.md:28]: done

File: CHANGELOG.md (hunk 8 above)
Comment (yhilmare) [reply CHANGELOG.md:7]: 看了下，以前的也都没有加句号，这个建议先不动吧～否则跟以前不统一了😂

**Review by yizhouxw — APPROVED** [review]

**Pull request merged**